spyder_unittest/widgets/tests/test_unittestgui.py

```diff
@@ -29,8 +29,8 @@ def test_run_tests_and_display_results(qtbot, tmpdir):
 
     widget = UnitTestWidget(None)
     qtbot.addWidget(widget)
-    widget.analyze(tmpdir.strpath)
-    qtbot.wait(10000)  # wait for tests to run
+    with qtbot.waitSignal(widget.sig_processed_tests, timeout=10000):
```

Comment:
Its good to add the parameter `raising=False` so that if for any reason this test is taking more than 10 seconds, it should fail (to warn us)